src/Recursion.php

```diff
@@ -14,4 +18,35 @@ public static function check(string $className, array $stack, string $modelPath)
 
         return false;
     }
+
+
+    public static function hasMaxDepthReached(PropertyMetadata $propertyMetadata, array $stack): bool
+    {
+        $className = self::getClassNameFromProperty($propertyMetadata);
+        if (null === $className) {
+            return false;
+        }
+
+        if (null === $propertyMetadata->getMaxDepth()) {
```

Comment:
should we do this check first? its probably cheaper than getting the class name - and it also semantically makes sense to not do anything if no max depth is defined.

Reply:
Good point, adjusted it via ad70b45bf28d2d1a41f22a7c2919be63ef31d150